go/storage/mkvs/checkpoint/restorer.go

```diff
@@ -117,7 +120,8 @@ func (rs *restorer) RestoreChunk(ctx context.Context, idx uint64, r io.Reader) (
 		rs.pendingChunks = nil
 		rs.currentCheckpoint = nil
 
-		return true, nil
+		err := rs.ndb.FinishMultipartInsert(false)
```

Comment:
Actually this needs to be deferred until `Finalize` is called as otherwise you could still leave nodes in the database if the process crashes between this call and `Finalize`.

I suggest you change the API a bit as follows:
* There is no `FinishMultipartInsert(bool)`, instead there is `AbortMultipartInsert()` which behaves the same as `FinishMultipartInsert(false)`.
* `StartMultipartInsert` is changed to take a version which is stored together with the `multipart` flag. All versions passed to `NewBatch`, `Commit` and `Finalize` calls must match the `multipartVersion` otherwise this is an error.
* If you want to finish the multipart insert successfully you need to call `Finalize` with the `multipartVersion`.
* Calling `Prune` while a multipart insert is in progress is an error.